src/iperf_udp.c

```diff
@@ -567,6 +652,34 @@ iperf_udp_connect(struct iperf_test *test)
 
     if (test->settings->varlen)
         srand(time(NULL));
+        
+    if (test->settings->lso_udp) {
+        int level, name, val;
+        uint16_t test_lso_segsize;
+        socklen_t tlsoss_size;
+
+        if (getsockopt(s, SOL_UDP, UDP_SEGMENT, &test_lso_segsize, &tlsoss_size)) {
+            i_errno = IEUDPSEGMENT;
+            test->settings->lso_udp = 0;
+            test->settings->lso_udp_segsize = 0;
+        }
+
+        if (test->settings->domain == PF_UNSPEC ||
+            test->settings->domain == PF_INET) {
+                level = SOL_IP;
+                name  = IP_MTU_DISCOVER;
+                val   = IP_PMTUDISC_DO;
+        } else if (test->settings->domain == PF_INET6) {
+                level = SOL_IPV6;
+                name  = IPV6_MTU_DISCOVER;
+                val   = IPV6_PMTUDISC_DO;
+        }
+
+        if (setsockopt(s, level, name, &val, sizeof(val))) {
```

Comment:
again, why you've added this? why you need MTU_DISCOVER for lso_udp specifically?